From ac5d9d48767f7a580d45f743afc902497adeb4d8 Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Sat, 18 Apr 2015 23:03:45 -0500
Subject: [PATCH 1/2] Make sure that that '/cgroup' and the controller are
 sep'd by /

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 lxcfs.c | 2 ++
 1 file changed, 2 insertions(+)

Index: lxcfs-0.7/lxcfs.c
===================================================================
--- lxcfs-0.7.orig/lxcfs.c
+++ lxcfs-0.7/lxcfs.c
@@ -313,6 +313,8 @@ static char *pick_controller_from_path(s
 
 	if (strlen(path) < 9)
 		return NULL;
+	if (*(path+7) != '/')
+		return NULL;
 	p1 = path+8;
 	ret = nih_strdup(NULL, p1);
 	if (!ret)
