From e5d26e0be93698fb99f81d2c6d472124a982a22d Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Sat, 18 Apr 2015 23:51:03 -0500
Subject: [PATCH 2/2] free d at program end

to silence warnings from memory checkers

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 lxcfs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: lxcfs-0.7/lxcfs.c
===================================================================
--- lxcfs-0.7.orig/lxcfs.c
+++ lxcfs-0.7/lxcfs.c
@@ -2393,7 +2393,7 @@ static bool is_help(char *w)
 int main(int argc, char *argv[])
 {
 	int ret;
-	struct lxcfs_state *d;
+	struct lxcfs_state *d = NULL;
 
 	if (argc < 2 || is_help(argv[1]))
 		usage(argv[0]);
@@ -2410,5 +2410,6 @@ int main(int argc, char *argv[])
 
 	ret = fuse_main(argc, argv, &lxcfs_ops, d);
 
+	free(d);
 	return ret;
 }
