Description: Make co-mounted controllers also show up separately
Author: Serge Hallyn <serge.hallyn@gmail.com>

Index: lxcfs-0.9/share/lxc.mount.hook.in
===================================================================
--- lxcfs-0.9.orig/share/lxc.mount.hook.in
+++ lxcfs-0.9/share/lxc.mount.hook.in
@@ -28,6 +28,16 @@ if [ -d "${LXC_ROOTFS_MOUNT}/sys/fs/cgro
 
             mkdir ${LXC_ROOTFS_MOUNT}/sys/fs/cgroup/$DEST
             mount -n --bind $entry ${LXC_ROOTFS_MOUNT}/sys/fs/cgroup/$DEST
+
+            # make sure that cpu,cpuacct shows up as cpu and cpuacct
+            # separately, else systemd is unhappy
+            if echo $DEST | grep -q ","; then
+                arr=$(echo $DEST | tr "," "\n")
+                for single in $arr
+                do
+                    ln -s $DEST ${LXC_ROOTFS_MOUNT}/sys/fs/cgroup/$single
+                done
+            fi
         done
 
         if [ -d ${LXC_ROOTFS_MOUNT}@LXCFSTARGETDIR@/ ]; then
